### PR TITLE
Attempt to fix postWrite hook, fix #138

### LINKED ISSUE #138: postWrite hook seems to interfere with creation of new user
Hi,

I think the maps app tries to read the exif information of the Coast.jpg file when the postWrite hook is being triggered. In my setup this file is empty at that moment. This causes the copyr operation for the skeleton to abort. On first login, users are welcomed with an "Internal Server Error". When logging in the second time it works but the only file in the Photos directory is the Coast.jpg with a size of 0.

This is logged when the user logs in the first time:
```
[index] Error: OCP\Lock\LockedException: "Photos/Coast.jpg" is locked at <<closure>>

 0. /var/www/nextcloud/lib/private/Files/View.php line 2050
    OC\Files\View->lockPath("/test2/files/Photos/Coast.jpg", 1, false)
 1. /var/www/nextcloud/lib/private/Files/View.php line 1140
    OC\Files\View->lockFile("/test2/files/Photos/Coast.jpg", 1)
 2. /var/www/nextcloud/lib/private/Files/View.php line 595
    OC\Files\View->basicOperation("file_get_contents", "/test2/files/Photos/Coast.jpg", ["read"])
 3. /var/www/nextcloud/lib/private/Files/Node/File.php line 52
    OC\Files\View->file_get_contents("/test2/files/Photos/Coast.jpg")
 4. /var/www/nextcloud/apps/maps/lib/Service/PhotofilesService.php line 447
    OC\Files\Node\File->getContent()
 5. /var/www/nextcloud/apps/maps/lib/Service/PhotofilesService.php line 391
    OCA\Maps\Service\PhotofilesService->getExifPelBackup("*** sensitive parameter replaced ***")
 6. /var/www/nextcloud/apps/maps/lib/Service/PhotofilesService.php line 278
    OCA\Maps\Service\PhotofilesService->getExif("*** sensitive parameter replaced ***")
 7. /var/www/nextcloud/apps/maps/lib/Service/PhotofilesService.php line 83
    OCA\Maps\Service\PhotofilesService->safeAddPhoto("*** sensitive parameter replaced ***", "*** sensitive parameter replaced ***")
 8. /var/www/nextcloud/apps/maps/lib/Hooks/FileHooks.php line 49
    OCA\Maps\Service\PhotofilesService->safeAddByFile("*** sensitive parameter replaced ***")
 9. <<closure>>
    OCA\Maps\Hooks\FileHooks->OCA\Maps\Hooks\{closure}("*** sensitive parameters replaced ***")
10. /var/www/nextcloud/lib/private/Hooks/EmitterTrait.php line 99
    undefinedundefinedcall_user_func_array(Closure {}, ["*** sensitive parameter replaced ***"])
11. /var/www/nextcloud/lib/private/Hooks/PublicEmitter.php line 36
    OC\Hooks\BasicEmitter->emit("\\OC\\Files", "postWrite", ["*** sensitive parameter replaced ***"])
12. /var/www/nextcloud/lib/private/Files/Node/Root.php line 136
    OC\Hooks\PublicEmitter->emit("\\OC\\Files", "postWrite", ["*** sensitive parameter replaced ***"])
13. /var/www/nextcloud/lib/private/Files/Node/Node.php line 109
    OC\Files\Node\Root->emit("\\OC\\Files", "postWrite", ["*** sensitive parameter replaced ***"])
14. /var/www/nextcloud/lib/private/Files/Node/File.php line 109
    OC\Files\Node\Node->sendHooks(["postWrite"])
15. /var/www/nextcloud/lib/private/legacy/util.php line 473
    OC\Files\Node\File->fopen("w")
16. /var/www/nextcloud/lib/private/legacy/util.php line 464
    OC_Util::copyr("/var/www/nextcloud/core/skeleton/Photos", OC\Files\Node\Folder {})
17. /var/www/nextcloud/lib/private/legacy/util.php line 436
    OC_Util::copyr("/var/www/nextcloud/core/skeleton", OC\Files\Node\Folder {})
18. /var/www/nextcloud/lib/private/User/Session.php line 520
    OC_Util::copySkeleton("*** sensitive parameter replaced ***", OC\Files\Node\Folder {})
19. /var/www/nextcloud/lib/private/User/Session.php line 378
    OC\User\Session->prepareUserLogin("*** sensitive parameter replaced ***", "*** sensitive parameter replaced ***")
20. /var/www/nextcloud/core/Controller/LoginController.php line 336
    OC\User\Session->completeLogin("*** sensitive parameters replaced ***")
21. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 166
    OC\Core\Controller\LoginController->tryLogin("*** sensitive parameters replaced ***")
22. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 99
    OC\AppFramework\Http\Dispatcher->executeController(OC\Core\Controller\LoginController {}, "tryLogin")
23. /var/www/nextcloud/lib/private/AppFramework/App.php line 126
    OC\AppFramework\Http\Dispatcher->dispatch(OC\Core\Controller\LoginController {}, "tryLogin")
24. /var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
    OC\AppFramework\App::main("OC\\Core\\Controller\\LoginController", "tryLogin", OC\AppFramework\ ... {}, {_route: "core.login.tryLogin"})
25. <<closure>>
    OC\AppFramework\Routing\RouteActionHandler->__invoke({_route: "core.login.tryLogin"})
26. /var/www/nextcloud/lib/private/Route/Router.php line 297
    undefinedundefinedcall_user_func(OC\AppFramework\ ... {}, {_route: "core.login.tryLogin"})
27. /var/www/nextcloud/lib/base.php line 975
    OC\Route\Router->match("/login")
28. /var/www/nextcloud/index.php line 42
    OC::handleRequest()

POST /index.php/login?clear=1
from 88.130.156.0 by test2 at 2019-09-06T20:05:37+00:00
```

For the system report refer to #116. I only updated to maps 0.1.2.

Please let me know if you need anything else.

### PR DESCRIPTION
**Pull request by tacruc**
Signed-off-by: Arne Hamann <kontakt+github@arne.email>

### CODE REVIEW

**Comment by alexthamm** [comment]
@tacruc the fix seems to be working. I'm able to create a new user and login as the user without any errors. The directory skeleton for the new user can been created successfully. All files are present and none of the files are 0 in size.

Since I'm not familiar with the Nextcloud/Maps app source code, my review may not be reliable: However I was wondering why you check the filesize instead of checking if the file is locked. I understand that the file locking mechanism is supposed to prevent files from being modified by multiple apps/users concurrently.

If for some reason an image is only partially written the filesize could be >0 while the file is still being locked. This should trigger the OCP\Lock\LockedException again.

**Comment by tacruc** [comment]
THX. Didn't think oft the locking will check this out tomorrow

**Comment by eneiluj** [comment]
Acquiring the `LOCK_EXCLUSIVE` does not work but `LOCK_SHARED` is fine. @tacruc if you're ok with my commit, I'll review this PR.

Thanks @alexthamm for the suggestion.

File: lib/Hooks/FileHooks.php [hunk 1]
```diff
@@ -45,11 +46,13 @@ public function __construct(IRootFolder $root, PhotofilesService $photofilesServ
 
     public function register() {
         $fileWriteCallback = function(\OCP\Files\Node $node) {
-            if ($this->isUserNode($node)) {
+            if ($this->isUserNode($node) && $node->getSize() > 0) {
```
Comment (tacruc) [inline lib/Hooks/FileHooks.php]: I think the comment from @alexthamm was more to add here an extra condition to check if the file is locked.

Comment (tacruc) [reply lib/Hooks/FileHooks.php]: maybe node->isReadable() ?

Comment (eneiluj) [reply lib/Hooks/FileHooks.php]: isReadable does not check if file is locked.

[OCP\Lock\ILockingProvider::isLocked](https://docs.nextcloud.com/server/16/developer_manual/api/OCP/Lock/ILockingProvider.html?highlight=islocked#OCP\Lock\ILockingProvider::isLocked) looks better, don't you think?

Comment (tacruc) [reply lib/Hooks/FileHooks.php]: That's what I was looking for

**Comment by eneiluj** [comment]
Done.

**Comment by tacruc** [comment]
I'm fine but can not review as it is my PR

**Pull request merged**